codec-http2/src/main/java/io/netty/handler/codec/http2/Http2LocalFlowController.java

```diff
@@ -66,4 +66,50 @@ void receiveFlowControlledFrame(ChannelHandlerContext ctx, Http2Stream stream, B
      * @return the number of unconsumed bytes for the stream.
      */
     int unconsumedBytes(Http2Stream stream);
+
+    /**
+     * Factory method to generate a new {@link LocalFlowState} object. This will be called after a new stream object
+     * is created and before {@link Http2Connection.Listener#onStreamAdded(Http2Stream)} is called. The result of this
+     * method will be returned by {@link Http2Stream#localFlowState()} for use by {@link Http2LocalFlowController}.
+     * @param stream The stream to associate the {@link LocalFlowState} with.
+     * This may be any stream, including the {@link Http2Connection#connectionStream()}.
+     * @return A new object which will represent local flow control state for {@code stream}.
+     */
+    LocalFlowState newLocalFlowState(Http2Stream stream);
+
+    /**
+     * A read only interface which represents the state required for the local flow control duties.
+     */
+    interface LocalFlowState {
+        /**
+         * The actual flow control window that is decremented as soon as {@code DATA} arrives.
+         */
+        int window();
+
+        /**
+         * A view of {@link #window()} that is used to determine when to send {@code WINDOW_UPDATE}
+         * frames. Decrementing this window for received {@code DATA} frames is delayed until the
+         * application has indicated that the data has been fully processed. This prevents sending
+         * a {@code WINDOW_UPDATE} until the number of processed bytes drops below the threshold.
+         */
+        int processedWindow();
+
+        /**
+         * This is what is used to determine how many bytes need to be returned relative to {@link #processedWindow()}.
+         */
+        int initialStreamWindowSize();
+
+        /**
+         * This is used to determine when {@link #processedWindow()} is sufficiently far away from
+         * {@link #initialStreamWindowSize} such that a {@code WINDOW_UPDATE} should be sent.
+         * Each stream has their own window update ratio.
+         */
+        float windowUpdateRatio();
```

Comment:
Some of these methods are definitely debatable as to weather they belong in this interface.  My approach was just to take most read only things we currently implement and put them in the interface.  I also added methods to help explain concepts in the comments.